code/game/objects/structures/satellite.dm

```diff
@@ -0,0 +1,47 @@
+/obj/structure/satellite
+	icon = 'icons/obj/machines/excelsior/objects.dmi'
+	icon_state = "sputnik"
+	var/cooldown = 0
+
+	attack_hand(mob/living/user as mob)
+		if(cooldown == 0)
+			cooldown = 1
+			emp_in(src.loc, 4, 6, 0)
+			sleep(360)
+			cooldown = 0
+
+/obj/structure/satellite/proc/emp_in(turf/epicenter, heavy_range, light_range, log=0)
```

Comment:
this seems like a straight up copypaste of generator's emp. Isn't there a literally proc for this?
`proc/empulse(turf/epicenter, heavy_range, light_range, log=0)`
In fact this seems like a copypaste of that too even.